systest/all_SUITE.erl

```diff
@@ -79,13 +79,27 @@ grisp_explode(Config) ->
     ),
     ok.
 
+compile_deps() ->
+    [{doc, "Issue #1712"
+           "When compile a project multiple time, the dependency should always be build event if refetch."}].
+compile_deps(Config) ->
+    rebar3("compile", Config),
+    rebar3("compile", Config),
+
+    PrivDir = ?config(path, Config),
+    BeansDir = filename:join([PrivDir, "_build", "default", "lib", "fake_dep", "ebin"]),
+
+    {ok, Beans} = file:list_dir(BeansDir),
+    ?assert(length(Beans) > 1).
+
+
 %%%%%%%%%%%%%%%
 %%% Helpers %%%
 %%%%%%%%%%%%%%%
 set_name_config(Atom, Config) ->
     [{path,
       filename:join([?config(priv_dir, Config),
-                     atom_to_list(?MODULE)++"_data", atom_to_list(Atom)])}
+                     atom_to_list(Atom)])}
```

Comment:
The directory was not created on my system. I was running on my mac. With the current path everything work as expected.

Reply:
This broke all the tests on my own end here. I'll see what is needed.

Reply:
This does patches tests on OSX though. Will be fun to figure out.

Reply:
okay I think there's some directory difference in path handling between OSX and linux and that's not good (since we can't detect this super well dynamically), but I'll keep on linux right now, since that's what used to work.